Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -155,14 +155,32 @@ template <class Type> class TableColumn : public API::Column {
    * is throw. In case of an overflow boost::numeric::positive_overflow or
    * boost::numeric::negative_overflow
    * is throw.
-   * @param i :: The index to an element.
+   * @param value :: The value of the element.
    */
-  double toDouble(size_t i) const override {
+  template <typename T> double convertToDouble(const T &value) const {
     typedef
```

Comment:
Ah yes I see. Can we then at least move to using the `std::` versions of the `type_traits` and remove the boost code? `boost::if_c` should be able to be replaced with [`std::conditional`](http://en.cppreference.com/w/cpp/types/conditional) and `boost::is_convertible` by `std::is_convertible.`